src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/IcebergFieldNode.java

```diff
@@ -1,54 +1,61 @@
 package com.snowflake.kafka.connector.internal.streaming.schemaevolution.iceberg;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
 import java.util.LinkedHashMap;
+import java.util.Map;
 import java.util.stream.Collectors;
 import org.apache.iceberg.types.Type;
 import org.apache.iceberg.types.Types;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
 
 class IcebergFieldNode {
 
-  public final String name;
+  // todo consider refactoring into some more classes
```

Comment:
i would split it into two classes at least. 
1. IcebergFieldNode - simple DTO class containing name, type and list of children
2. Some service/bean class responsible for creating and merging IcebergFieldNode